tests/integration/test_storage_delta/test.py

```diff
@@ -1024,3 +1070,84 @@ def test_replicated_database_and_unavailable_s3(started_cluster, use_delta_kerne
         )
+
+
+def test_session_token(started_cluster):
+    spark = started_cluster.spark_session
+    minio_client = started_cluster.minio_client
+    TABLE_NAME = randomize_table_name("test_session_token")
+    bucket = started_cluster.minio_bucket
+
+    if not minio_client.bucket_exists(bucket):
+        minio_client.make_bucket(bucket)
+
+    node_name = "node_with_environment_credentials"
+    instance = started_cluster.instances[node_name]
+    parquet_data_path = create_initial_data_file(
+        started_cluster,
+        instance,
+        "SELECT toUInt64(number), toString(number) FROM numbers(100)",
+        TABLE_NAME,
+        node_name=node_name,
+    )
+
+    write_delta_from_file(spark, parquet_data_path, f"/{TABLE_NAME}")
+    upload_directory(minio_client, bucket, f"/{TABLE_NAME}", "")
+
+    assert 0 < int(
+        instance.query(
+            f"""
+    SELECT count() FROM deltaLake(
+        'http://{started_cluster.minio_host}:{started_cluster.minio_port}/{started_cluster.minio_bucket}/{TABLE_NAME}/',
+        SETTINGS allow_experimental_delta_kernel_rs=1)
+    """
+        )
+    )
+
+    instance2 = started_cluster.instances["node1"]
+
+    assert 0 < int(
+        instance2.query(
+            f"""
+    SELECT count() FROM deltaLake(
+        'http://{started_cluster.minio_host}:{started_cluster.minio_port}/{started_cluster.minio_bucket}/{TABLE_NAME}/', 'minio', 'minio123',
+        SETTINGS allow_experimental_delta_kernel_rs=1)
+    """
+        )
+    )
+
+    assert (
+        "Received DeltaLake kernel error ReqwestError: Error interacting with object store"
+        in instance2.query_and_get_error(
+            f"""
+    SELECT count() FROM deltaLake(
+        'http://{started_cluster.minio_host}:{started_cluster.minio_port}/{started_cluster.minio_bucket}/{TABLE_NAME}/', 'minio', 'minio123', 'fake-token',
+        SETTINGS allow_experimental_delta_kernel_rs=1)
+    """
+        )
+    )
+
+    instance3 = started_cluster.instances["node_with_session_token"]
+
+    assert 0 < int(
+        instance3.query(
+            f"""
+       SELECT count() FROM deltaLake(
+           'http://{started_cluster.minio_host}:{started_cluster.minio_port}/{started_cluster.minio_bucket}/{TABLE_NAME}/',
+           SETTINGS allow_experimental_delta_kernel_rs=1)
+       """
+        )
+    )
+
+    #expected_logs = [
+    #    "Calling EC2MetadataService to get token",
+    #    "Calling EC2MetadataService resource, /latest/meta-data/iam/security-credentials with token returned profile string myrole",
+    #    "Calling EC2MetadataService resource resolver:8080/latest/meta-data/iam/security-credentials/myrole with token",
+    #    "Successfully pulled credentials from EC2MetadataService with access key",
+    #]
+
+    #instance3.query("SYSTEM FLUSH LOGS")
+    #for expected_msg in expected_logs:
+    #    assert instance3.contains_in_log(
+    #        "AWSEC2InstanceProfileConfigLoader: " + expected_msg
+    #    )
```

Comment:
Should it be removed or uncommented?

Reply:
Fixed in [4aedb21](https://github.com/ClickHouse/ClickHouse/pull/77661/commits/4aedb219ec4ae5f0068cdb45e4470e55eb693a80)